src/cloud_what/providers/gcp.py

```diff
@@ -165,11 +170,45 @@ def _get_metadata_from_server(self) -> Union[str, None]:
         :return: String with token or None
         """
         token = self._get_data_from_server(data_type="token", url=self.CLOUD_PROVIDER_METADATA_URL)
+
         if token is not None:
+            ttl = self.CLOUD_PROVIDER_TOKEN_TTL
+            ctime = time.time()
+
+            # Try to get expiration time from JWT token and the time, when the token was created
+            jose_header, metadata, encoded_signature = self.decode_jwt(token)
+            if metadata is not None:
+                try:
+                    metadata = json.loads(metadata)
+                except json.JSONDecodeError as err:
+                    log.error(f"Unable to decode metadata from JWT token: {err}")
+                    metadata = {}
+                if "exp" in metadata:
+                    exp_time = metadata["exp"]
+                    if "iat" in metadata:
+                        # We try to test if there is not too big difference between creation time of JWT
+                        # token and current time (more than 10 seconds), because it would be suspicious,
+                        # and it could cause some other issues
+                        iat_time = metadata["iat"]
+                        if abs(ctime - iat_time) > self.THRESHOLD:
+                            log.warning(
+                                f"Too big diff between ctime: {ctime} " "and GCP JWT token iat: {iat_time}"
+                            )
```

Comment:
I will change the comment. Such situation would IMHO suggest that there is something wrong with e.g. NTP server used by client. It doesn't suggest that JWT token is suspicious.

BTW: JWT token contains signature, but we just cannot check it, because it would require too many dependencies.

Reply:
Updated the comment and refactored the code.

Reply:
Added unit test.